src/Util.vala

```diff
@@ -166,23 +166,55 @@ namespace Tasks.Util {
     //--- X-Property ---//
 
 
-    public unowned ICal.Property? get_icalcomponent_x_property (ICal.Component ical_component, string x_property_name) {
+    public ICal.Property? get_icalcomponent_x_property (ICal.Component ical_component, string x_property_name) {
         return get_ecalpropertybag_x_property (new ECal.ComponentPropertyBag.from_component (ical_component, (property) => {
             return property.isa () == ICal.PropertyKind.X_PROPERTY;
         }), x_property_name);
     }
 
-    public unowned ICal.Property? get_ecalpropertybag_x_property (ECal.ComponentPropertyBag ecal_propertybag, string x_property_name) {
-        unowned ICal.Property? property = null;
-
+    public ICal.Property? get_ecalpropertybag_x_property (ECal.ComponentPropertyBag ecal_propertybag, string x_property_name) {
         var ecal_propertybag_count = ecal_propertybag.get_count ();
+
         for (int i = 0; i < ecal_propertybag_count; i++) {
-            property = ecal_propertybag.get (i);
+            unowned ICal.Property? property = ecal_propertybag.get (i);
             if (property.isa () == ICal.PropertyKind.X_PROPERTY && property.get_x_name () == x_property_name) {
-                break;
+                var x_property = new ICal.Property (ICal.PropertyKind.X_PROPERTY);
+                x_property.set_x_name (x_property_name);
+
+                if (property.get_x () != null) {
+                    x_property.set_x (property.get_x ().dup ());
+                }
+
+                return x_property;
             }
         }
-        return property;
+        return null;
+    }
+
+
+    /**
+     * Returns the value of X-APPLE-SORT-ORDER property if set
+     */
+    public string? get_apple_sortorder_property_value (ECal.Component ecalcomponent) {
+        unowned ICal.Component? icalcomponent = ecalcomponent.get_icalcomponent ();
+        if (icalcomponent != null) {
+            var x_apple_sort_order_property = get_icalcomponent_x_property (icalcomponent, "X-APPLE-SORT-ORDER");
+
+            if (x_apple_sort_order_property != null ) {
+                return x_apple_sort_order_property.get_x ();
+            }
+        }
+        return null;
+    }
+
+    /**
+    * if the X-APPLE-SORT-ORDER property is not set,
+    * Apple uses the number of seconds between the creation of
+    * the task and 20010101T000000Z (Cocoa/Webkit epoch)
+    * as value for sorting.
+    */
+    public ICal.Duration get_apple_sortorder_default_value (ECal.Component ecalcomponent) {
+        return ecalcomponent.get_created ().subtract (new ICal.Time.from_string ("20010101T000000Z"));
     }
```

Comment:
What's the benefit of this over using the old comparison method? I'm not a huge fan of magic numbers

Reply:
I'm not a huge fan of this either - but in `ListView.sort_function`, we need something to compare against, in case `sortorder_a` is set but `sortorder_b` isn't: If we use the values as before, all tasks with `dtstart` or `dtstamp` set (but no `x-apple-sort-order`) will be shown before the manually sorted ones (the ones with `x-apple-sort-order` set) - therefore mainly defeating the purpose of manual sorting.

So either I'm missing something here, or your question boils down to whether we should implement custom sorting or not. And to be more specific here: The Apple way of custom sorting.

Because there is no official way for custom sorting in the CalDAV standard, we need to come up with our own solution - if we want to provide this feature. The Apple method for custom sorting has the added benefit by already being implemented in a bunch of software such as Nextcloud, iOS (and potentially others).

Reply:
FWIW: I tested with using `row_a.task.get_created ().as_timet ().to_string ()` instead of the Apple magic value in case the property value is not available. While technically it works, it leads to different results than on the iOS device. However, we can go this route and re-implement the Apple default behavior again later on if there is an actual need for it?

Reply:
Ah okay, I think I just wasn't understanding how all these pieces fit together. Maybe can you amend the comment with a brief explanation that we're doing this so that we can match the sort order on iOS devices?